book/sending_emails.rst

```diff
@@ -0,0 +1,226 @@
+Sending E-Mails
+===============
+
+Creating an E-Mail
+------------------
+
+An e-mail that is created and sent by the OroPlatform is centered around the
+:class:`Oro\\Bundle\\EmailBundle\\Form\\Model\\Email` model class which reflects all the properties
+of an e-mail.
+
+There are two ways to create a new e-mail:
+
+#. :ref:`Manually create an e-mail <book-sending-emails-manually>`.
+
+#. :ref:`Use the e-mail builder class <book-sending-emails-builder>`.
+
+.. _book-sending-emails-manually:
+
+Manually Creating an E-Mail
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can manually create an e-mail by creating a new instance of the ``Email`` model class and call
+the setter methods for all the properties you want to be set:
+
+.. code-block:: php
+    :linenos:
+
+    // src/Acme/DemoBundle/Controller/EmailController.php
+    namespace Acme\DemoBundle\Controller;
+
+    use Oro\Bundle\EmailBundle\Entity\EmailTemplate;
+    use Oro\Bundle\EmailBundle\Form\Model\Email;
+    use Oro\Bundle\EmailBundle\Form\Model\EmailAttachment;
+    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+    class EmailController extends Controller
+    {
+        public function sendMailAction()
+        {
+            $email = new Email();
+
+            // the sender
+            $email->setFrom('chris@example.com');
+
+            // recipient(s)
+            $email->setTo(array('alice@example.com', 'bob@example.com'));
+
+            // CC recipient(s)
+            $email->setCc(array('dave@example.com', 'eric@example.com'));
+
+            // BCC recipient(s)
+            $email->setBcc(array('ryan@example.com', 'jonathan@example.com'));
+
+            // the subject
+            $email->setSubject(...);
+
+            // a template to create the e-mail body, the passed template
+            // must be an instance of Oro\Bundle\EmailBundle\Entity\EmailTemplate
+            $email->setTemplate(...);
+
+            // a context that will be passed to the template set with setTemplate()
+            $email->setContexts(array(...));
+
+            // the e-mail type (either html or text)
+            $email->setType(...);
+
+            // as an alternative to using a template (see above) you can also
+            // directly set the e-mail body
+            $email->setBody(...);
+
+            // e-mail attachments (instances of  Oro\Bundle\EmailBundle\Form\Model\EmailAttachment
+            $email->addAttachment($attachment);
+
+            // ...
+        }
+    }
+
+.. _book-sending-emails-builder:
+
+
+The ``EmailModelBuilder`` Class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An alternative approach to manually creating ``Email`` models is to use the
+:class:`Oro\\Bundle\\EmailBundle\\Builder\\EmailModelBuilder` helper class which offers several
+methods to create new e-mails based on existing data:
+
+:method:`Oro\\Bundle\\EmailBundle\\Builder\\EmailModelBuilder::createEmailModel`
+    Create an e-mail as a copy of an existing e-mail.
```

Comment:
Create new or fills the additional data only. Do not copy of an existing e-mail.